testcontainers/testcontainers-go
# feat (postgres): support for creating and restoring Snapshots

## What does this PR do?



I recently went through building a complex testing setup for an entire webservice using testcontainers-go, which taught me some important tips and tricks about the library. I have added these in the documentation wherever I thought they could be useful.

I also added a new example for using the postgres module and making each test use a clean database. It's a pretty useful tool, and it's available as an example in the Java version, so I've added it here as well.

## Why is it important?



During my own work, I ran into some roadblocks where the docs did not offer all the information I needed to get things working. I worked things with the team and by reading the official Java example. I thought it'd be a good idea to upstream those learnings to the library.

## Related issues


- 

## How to test this PR

Run the postgres example, it should pass and should show how we can use a clean database in multiple tests.
